pmm-app/src/pmm-pt-summary/datasource/PTSummaryDataSource.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  DataQueryResponse,
+  DataSourceApi,
+  MutableDataFrame,
+  FieldType,
+  DataSourceInstanceSettings,
+} from '@grafana/data';
+import { PTSummaryService } from './PTSummary.service';
+
+/* eslint-disable no-useless-constructor, class-methods-use-this */
+export class PTSummaryDataSource extends DataSourceApi {
+  constructor(instanceSettings: DataSourceInstanceSettings) {
```

Comment:
Do we really need this constructor? Hope not :)

Reply:
Yes, it's required, I tried removing it and Grafana is not able to create the Datasource anymore. Makes sense because DataSourceApi is abstract.

Reply:
Well, given that there are no abstract types in JS, I wonder what causes this behaviour? Typescript?

Reply:
Exactly that. Typescript has abstract keyword and behavior.